app/services/problem_check/ai_llm_status.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class ProblemCheck::AiLlmStatus < ProblemCheck
+  self.priority = "high"
+  self.perform_every = 6.hours
+
+  def call
+    llm_errors
+  end
+
+  private
+
+  def llm_errors
+    return [] if !SiteSetting.discourse_ai_enabled
+
+    LlmModel.in_use.find_each.filter_map do |model|
+      try_validate(model) { validator.run_test(model) }
+    end
+  end
+
+  def try_validate(model, &blk)
+    begin
+      raise({ message: "Forced error for testing" }.to_json) if Rails.env.test?
```

Comment:
Hmm... this means you can't test the use case for "model is enabled and working fine" though.

Reply:
Does the dummy model have an auth mechanism? One way I've seen this tested in other places (login methods for example) is by mocking the author response, if it doesn't return a 200, then the problem check is raised.

Reply:
In other words, if the credentials are wrong, will the problem check be raised here?

Reply:
You will have to config a real model and stub an HTTP request to simulate the error case. Set one of the module's site setting to that model and run the test.